### ISSUE TITLE
Unable to initialize physfs on Android platform

### ISSUE THREAD

**Issue by thehatkid** [icculus/physfs#52]
Hello, I want to use PhysicsFS 3.2.0 (with SDL2) on Android, I using JNI C++ native for my Android application, and I can't initialize with `argv[0]`, it just crashes with segmentation fault (SIGSEGV) error.

However, passing `NULL` as first argument to `PHYSFS_init()` is working fine, but with no base directory (`PHYSFS_getBaseDir()` returns `/`)

In debugging, I have found the issue, and it's in `physfs_platform_android.c` at method `__PHYSFS_platformCalcBaseDir` in line 57:
https://github.com/icculus/physfs/blob/9266e773d34c341877bd32671d674df20c6fd904/src/physfs_platform_android.c#L57

My part of code where I initing the PhysFS:
```cpp
#include <SDL.h>
#include <physfs.h>
#include <android/log.h>

int SDL_main(int argc, char *argv[])
{
	__android_log_print(ANDROID_LOG_INFO, "hksoft", "Hello, Android!");
	__android_log_print(ANDROID_LOG_INFO, "hksoft", "ARGC is: %d, ARGV[0] is: %s", argc, argv[0]); // ARGC is: 1, ARGV[0] is: app_process

	// Init PhysFS
	if (PHYSFS_init(argv[0])) { // <-- Segmentation fault
		__android_log_print(ANDROID_LOG_INFO, "PHYSFS", "Init Successful!");
	} else {
		__android_log_print(ANDROID_LOG_FATAL, "PHYSFS", "Init Failure! Error: %s", PHYSFS_getLastError());
		return 1;
	}

	// Log some PhysFS stuff (if PhysFS was inited successfully)
	const char *baseDir = PHYSFS_getBaseDir(); // /
	const char *userDir = PHYSFS_getUserDir(); // /data/
	const char *prefDir = PHYSFS_getPrefDir("hat_kid", "SomethingGame"); // (null)
	const char *writeDir = PHYSFS_getWriteDir(); // (null)
	__android_log_print(ANDROID_LOG_INFO, "PHYSFS", "Base directory is: %s", baseDir); // Base directory is: /
	__android_log_print(ANDROID_LOG_INFO, "PHYSFS", "User directory is: %s", userDir); // User directory is: /data/
	__android_log_print(ANDROID_LOG_INFO, "PHYSFS", "Pref directory is: %s", prefDir); // Pref directory is: (null)
	__android_log_print(ANDROID_LOG_INFO, "PHYSFS", "Write directory is: %s", writeDir); // Write directory is: (null)

	// Init SDL2
	SDL_Init(SDL_INIT_VIDEO);

	...

	// Clean up
	SDL_Quit();

	return 0;
}
```

Screenshot of ADB logcat log:
![Screenshot of ADB logcat log](https://user-images.githubusercontent.com/35992897/212465787-0ca7c260-6684-4e37-90d9-7cf5ef126dbd.png)

Android version is 9, arch is armeabi-v7a (32-bit).

**Comment by thehatkid** [comment]
Okay, I got it working PhysFS 3.2.0 by reading the warning in method `PHYSFS_init` in the header file (`physfs.h`), so thanks anyway!
I surprised what I can't find about this issue on internet, so I will write solution here. I hope it would help.

I think it would change the initilization on Android platform in PhysFS 4.0 anyway (I have read this from plans in issues tab)

The **solution** is putting `PHYSFS_AndroidInit` struct instead of `argv[0]` to `PHYSFS_init()` first argument:
```cpp
JNIEnv *jenv = (JNIEnv *)SDL_AndroidGetJNIEnv();
jobject jctx = getJNIGlobalContext(jenv); // https://stackoverflow.com/a/46871051/13295652

PHYSFS_AndroidInit ainit;
ainit.jnienv = jenv;
ainit.context = jctx;

PHYSFS_init((char *)&ainit);

...
```
